minix/commands/autopart/autopart.c

```diff
@@ -1531,8 +1531,13 @@ void m_read(int ev, int *biosdrive)
 	stat_start(0);
 	fflush(stdout);
 
-	if ((device= open(curdev->name, mode= O_RDWR, 0666)) < 0) {
-		if (device >= 0) { close(device); device= -1; }
+	device = open(curdev->name, O_RDWR, 0666);
+	if (device < 0) {
+		return;
+	}
+	else if (device >= 0) {
+		close(device);
+		device = -1;
```

Comment:
Oops! I've removed the else if block.
Do you want me to put it back to what is was before but without line 1535? It's only a couple extra lines and it is, IMO, much easier to read and understand.